pkg/cleanup/services.go

```diff
@@ -0,0 +1,62 @@
+package cleanup
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/k0sproject/k0s/pkg/install"
+	"github.com/kardianos/service"
+	"github.com/sirupsen/logrus"
+)
+
+type services struct {
+	Config *Config
+	roles  []string
+}
+
+func (s *services) Name() string {
+	return "uninstal service step"
+}
+func (s *services) NeedsToRun() bool {
+	possibleRoles := []string{
+		"controller", "worker",
+	}
+	for _, prole := range possibleRoles {
+		if _, stub, err := install.GetSysInit(prole); err == nil && stub != "" {
+			s.roles = append(s.roles, prole)
+		}
+	}
+
+	return len(s.roles) > 0
+}
+
+func (s *services) Run() error {
+	var msg []string
+	for _, role := range s.roles {
+		if err := uninstallService(role); err != nil {
+			logrus.Debugf("Tried removing service: %v", err)
+			msg = append(msg, err.Error())
+		}
+	}
+	if len(msg) > 0 {
+		return fmt.Errorf("%v", strings.Join(msg, "\n"))
+	}
+	return nil
+}
+
+func uninstallService(role string) error {
+	prg := &install.Program{}
+
+	if role == "controller+worker" {
```

Comment:
I don't think this check is really needed. This func get called only from the above `Run()` and thus should be only called for `controller` or `worker` roles.